processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/FragmentCompletionTimeDeriver.java

```diff
@@ -90,4 +95,16 @@ public CompletionTime processOneToOne(String tenantId, Span item) throws RetryAt
         return null;
     }
 
+    private void setFault(CompletionTime completionTime, List<BinaryAnnotation> binaryAnnotations) {
+        List<HttpCodesUtil.HttpCode> errorCodes =
+                HttpCodesUtil.getClientOrServerErrors(HttpCodesUtil.getHttpStatusCodes(binaryAnnotations));
+
+        if (errorCodes.size() > 0) {
```

Comment:
One HTTP request can get only one HTTP response, and an HTTP response can have only one HTTP status code. It's a bit confusing to me, as I was under the impression that you might be dealing with more than one HTTP request here. If that's not the case at all, then you are having a code that is more complex than it should, for the sake of possible future need. I'd say that if you need that in the future, then you can refactor that in the future.